### PR TITLE
Sean---LA114.Wind

### PR DESCRIPTION
**Pull request by swd-turner**
#142

### CODE REVIEW

**Comment by swd-turner** [comment]
@gokuliyer when reviewing, please check units on output file. The old data system has it as $/GJ, but i'm pretty sure it's a unitless capacity factor.

File: R/zchunk_LA114.wind-gcamusa.R [hunk 1]
```diff
@@ -1,70 +1,83 @@
 #' module_gcam.usa_LA114.Wind
 #'
-#' Briefly describe what this chunk does.
+#' Computes capacity factors for wind by US state.
```
Comment (bpbond) [inline R/zchunk_LA114.wind-gcamusa.R]: Unbelievably picky, but: "Compute"

File: R/zchunk_LA114.wind-gcamusa.R [hunk 2]
```diff
@@ -1,70 +1,83 @@
...
+    # This function filters A23 tables for wind, then gathers and interpolates...
+    # ... to get a single value for wind base cost year. Note that the interpolation is...
+    # ... redundant whilst gcamusa.WIND_BASE_COST_YEAR = 2005, ...
+    # ... since 2005 is an existing data point in all A23 tables.
+    filter_gather_interp_get_cost <- function(x){
+      x %>% filter(technology == "wind") %>%
+        gather(year,value, `1971`, `1990`, `2005`, `2010`, `2015`) %>%
+        select(year,value) %>% mutate(year = as.integer(year)) %>%
+        complete(year = c(year, gcamusa.WIND_BASE_COST_YEAR)) %>%
+        mutate(value = approx_fun(year, value, rule = 2)) -> x_
```
Comment (bpbond) [inline R/zchunk_LA114.wind-gcamusa.R]: No need to end with `return`:
```R
mutate(value = approx_fun(year, value, rule = 2)) %>%
filter(x_, year == gcamusa.WIND_BASE_COST_YEAR)$value
```

File: R/zchunk_LA114.wind-gcamusa.R [hunk 3]
```diff
@@ -1,70 +1,83 @@
...
+      return(filter(x_, year == gcamusa.WIND_BASE_COST_YEAR)$value)
+    }
+
+    A23.globaltech_capital %>% filter_gather_interp_get_cost -> L114.CapCost
+    A23.globaltech_OMfixed %>% filter_gather_interp_get_cost -> L114.OMFixedCost
+    A23.globaltech_OMvar %>% filter_gather_interp_get_cost -> L114.OMVarCost
+    # ^^ all above rates are in $1975
+
+    # Get fixed charge rate of capital for wind
+    filter(A23.globaltech_capital, technology=="wind")$fixed.charge.rate -> L114.FixedChargeRate
```
Comment (bpbond) [inline R/zchunk_LA114.wind-gcamusa.R]: Spaces around `==` please

File: R/zchunk_LA114.wind-gcamusa.R [hunk 4]
```diff
@@ -1,70 +1,83 @@
...
+
+    # Convert state level wind base cost data to 1975$/GJ, ...
+    # ... then compute capacity factor for the base wind turbine in each state
+    us_state_wind %>%
+      mutate(sector = "electricity generation",
+             fuel = "wind",
+             base_cost = base_cost * gcamdata::gdp_deflator(1975, 2007) / CONV_KWH_GJ,
```
Comment (bpbond) [inline R/zchunk_LA114.wind-gcamusa.R]: Note the `gcamdata::` prefix is unnecessary. But it does clarify the source of this function, so can leave it if you want.

File: R/zchunk_LA114.wind-gcamusa.R [hunk 5]
```diff
@@ -1,70 +1,83 @@
...
-      add_comments("comments describing how data generated") %>%
-      add_comments("can be multiple lines") %>%
+    # This function filters A23 tables for wind, then gathers and interpolates...
+    # ... to get a single value for wind base cost year. Note that the interpolation is...
+    # ... redundant whilst gcamusa.WIND_BASE_COST_YEAR = 2005, ...
+    # ... since 2005 is an existing data point in all A23 tables.
+    filter_gather_interp_get_cost <- function(x){
+      x %>% filter(technology == "wind") %>%
+        gather(year,value, `1971`, `1990`, `2005`, `2010`, `2015`) %>%
+        select(year,value) %>% mutate(year = as.integer(year)) %>%
```
Comment (bpbond) [inline R/zchunk_LA114.wind-gcamusa.R]: `year, value` in these two lines

File: R/zchunk_LA114.wind-gcamusa.R [hunk 6]
```diff
@@ -1,70 +1,83 @@
...
-      add_units("units") %>%
-      add_comments("comments describing how data generated") %>%
-      add_comments("can be multiple lines") %>%
+    # This function filters A23 tables for wind, then gathers and interpolates...
+    # ... to get a single value for wind base cost year. Note that the interpolation is...
+    # ... redundant whilst gcamusa.WIND_BASE_COST_YEAR = 2005, ...
+    # ... since 2005 is an existing data point in all A23 tables.
+    filter_gather_interp_get_cost <- function(x){
+      x %>% filter(technology == "wind") %>%
+        gather(year,value, `1971`, `1990`, `2005`, `2010`, `2015`) %>%
```
Comment (bpbond) [inline R/zchunk_LA114.wind-gcamusa.R]: `gather(year, value, matches(YEAR_PATTERN))` ?

**Comment by bpbond** [comment]
Thanks, Gokul. @swd-turner this is waiting on minor tweaks from you.

**Pull request merged**